flocx_market/api/offer_contract_relationship.py

```diff
@@ -6,34 +6,56 @@
 class OfferContractRelationship(Resource):
 
     @classmethod
-    def get(cls, marketplace_offer_id=None, contract_id=None):
-        if (contract_id is None) and (marketplace_offer_id is None):
+    def get(cls):
+        ocr_id = request.args.get('offer_contract_relationship_id')
+        if ocr_id:
+            ocr_with_id = ocr.OfferContractRelationship.get(ocr_id)
+            return ocr_with_id.to_dict()
+
+        marketplace_offer_id = request.args.get('marketplace_offer_id')
+        contract_id = request.args.get('contract_id')
+        status = request.args.get('status')
+
+        possible_filters = {
+            'marketplace_offer_id': marketplace_offer_id,
+            'contract_id': contract_id,
+            'status': status
+        }
+        filters = {}
+        for key, value in possible_filters.items():
+            if value is not None:
+                filters[key] = value
+
+        if len(filters) == 0:
             return [x.to_dict() for x in
                     ocr.OfferContractRelationship.get_all()]
-        ocrs = ocr.OfferContractRelationship.get(marketplace_offer_id,
-                                                 contract_id)
+        ocrs = ocr.OfferContractRelationship.get_all(**filters)
```

Comment:
Will this return an empty list if no matches without throwing an exception? That should be allowed.

Reply:
It should, I'll write a test for it